src/GoTrueClient.ts

```diff
@@ -601,6 +605,17 @@ export default class GoTrueClient {
     }
   }
 
+  private _isValidSession(maybeSession: unknown): maybeSession is Session {
+    const isValidSession =
+      typeof maybeSession === 'object' &&
+      maybeSession !== null &&
+      'access_token' in maybeSession &&
+      'refresh_token' in maybeSession &&
+      'expires_at' in maybeSession
```

Comment:
Is this check sufficient enough to ensure the data in `localStorage` is actually a session?

Reply:
hmm why do we need to check if the data in `localStorage` is a valid session? i think a valid session should be defined as one that has a valid refresh token and an access token that hasn't expired yet

Reply:
We need to check as after people upgrade, some people might have the old session format stored in `localStroage`, which will break us trying to read in that session. This check is just to make sure that what's there is probably a session

Reply:
hmm would renaming it to `_doesSessionExist` make it clearer?

Reply:
Good call – renaming!